core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -466,7 +466,7 @@ object SparkSubmit extends CommandLineUtils {
       // Other options
       OptionAssigner(args.executorCores, STANDALONE | YARN, ALL_DEPLOY_MODES,
         sysProp = "spark.executor.cores"),
-      OptionAssigner(args.executorMemory, STANDALONE | MESOS | YARN, ALL_DEPLOY_MODES,
+      OptionAssigner(args.executorMemory, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
```

Comment:
If this was the only change then specifying `local-cluster[2,1,2048]` doesn't actually do anything because we're not setting `spark.executor.memory` anywhere. You could do `--master local-cluster[2,1,2048] --conf spark.executor.memory=2g` but that's cumbersome and now there are two ways to set the executor memory.